pageserver/src/tenant/timeline.rs

```diff
@@ -2503,65 +2493,8 @@ impl Timeline {
         frozen_layer: Arc<InMemoryLayer>,
         ctx: &RequestContext,
     ) -> Result<(), FlushLayerError> {
-        // As a special case, when we have just imported an image into the repository,
-        // instead of writing out a L0 delta layer, we directly write out image layer
-        // files instead. This is possible as long as *all* the data imported into the
-        // repository have the same LSN.
         let lsn_range = frozen_layer.get_lsn_range();
-        let (layers_to_upload, delta_layer_to_add) =
-            if lsn_range.start == self.initdb_lsn && lsn_range.end == Lsn(self.initdb_lsn.0 + 1) {
```

Comment:
I think the end check was ruining it.

Reply:
After discussing this more widely, I started to think perhaps the flushing should had always just started after importing all of the images and before importing wal, or done unconditionally after both. Then if the end requirement was left unchecked, we would always get the single image-only layer of layers.